client/my-sites/themes/validate-filters.js

```diff
@@ -30,5 +31,21 @@ module.exports = function validateFilter( context, next ) {
 	}
 
 	next();
-};
+}
 
+export function validateVertical( context, next ) {
+	const { vertical } = context.params;
+
+	if ( ! vertical ) {
+		return next();
+	}
+
+	if ( ! includes( getSubjects(), vertical ) ) {
+		if ( context.isServerSide ) {
+			return context.res.redirect( '/themes' );
+		}
+		return page.redirect( '/themes' );
```

Comment:
I tried the following (re-using the theme sheet 404 for now):

```diff
diff --git a/client/my-sites/themes/index.node.js b/client/my-sites/themes/index.node.js
index d6a34cb..aa62c49 100644
--- a/client/my-sites/themes/index.node.js
+++ b/client/my-sites/themes/index.node.js
@@ -11,6 +11,7 @@ import {
        redirectFilterAndType,
        redirectToThemeDetails
 } from './controller';
+import { notFoundError } from '../theme/controller';
 import { validateFilters, validateVertical } from './validate-filters';
 
 export default function( router ) {
@@ -69,5 +70,6 @@ export default function( router ) {
                ], redirectToThemeDetails );
                // The following route definition is needed so direct hits on `/themes/<mysite>` don't result in a 404.
                router( '/themes/*', fetchThemeData, loggedOut, makeLayout );
+               router( notFoundError );
        }
 }
diff --git a/client/my-sites/themes/validate-filters.js b/client/my-sites/themes/validate-filters.js
index 77f33f4..49b0d5f 100644
--- a/client/my-sites/themes/validate-filters.js
+++ b/client/my-sites/themes/validate-filters.js
@@ -44,7 +44,11 @@ export function validateVertical( context, next ) {
                if ( context.isServerSide ) {
                        return context.res.redirect( '/themes' );
                }
-               return page.redirect( '/themes' );
+               const err = {
+                       status: 404,
+                       message: 'Theme Vertical Doesn\'t Exist',
+               };
+               return next( err );
        }
 
        next();
```

And that does seem to break the  `/themes/mood` redirect.